build-support/bin/native/bootstrap.sh

```diff
@@ -71,22 +72,30 @@ function _ensure_build_prerequisites() {
   export CARGO_HOME=${CACHE_ROOT}/rust-toolchain
   export RUSTUP_HOME=${CARGO_HOME}
 
+  local rust_toolchain="1.20.0"
+
   if [[ ! -x "${RUSTUP_HOME}/bin/rustup" ]]
   then
     log "A pants owned rustup installation could not be found, installing via the instructions at" \
         "https://www.rustup.rs ..."
     local readonly rustup=$(mktemp -t pants.rustup.XXXXXX)
     curl https://sh.rustup.rs -sSf > ${rustup}
-    sh ${rustup} -y --no-modify-path 1>&2
+    sh ${rustup} -y --no-modify-path --default-toolchain "${rust_toolchain}" 1>&2
     rm -f ${rustup}
-    ${RUSTUP_HOME}/bin/rustup override set stable 1>&2
   fi
 
+  # Make sure rust is pinned at the correct version.
+  # Check the current version, rather than just setting the version,
+  # because rustup outputs some text even if it's a no-op, and it's a little spammy.
+  # It would be nice if `rustup show` had some way of being usefully parsable, but it's not, so instead look at `rustc --version`.
+  # We sincerely hope that no one ever runs `rustup override set` in a subdirectory of the working directory.
+  [[ "$("${CARGO_HOME}/bin/rustc" --version | cut -d' ' -f2)" == "${rust_toolchain}" ]] || "${CARGO_HOME}/bin/rustup" override set "${rust_toolchain}" >&2
```

Comment:
I find a real time of ~0.04s in the no-op case for `rustup default 1.20.0`; ie maybe don't check the version and always just set the default, which will download it if needed.

Reply:
Done, means a little more logspam, but not the end of my world.